python/cudf/cudf/core/udf/masked_typing.py

```diff
@@ -411,6 +411,29 @@ def generic(self, args, kws):
             return nb_signature(MaskedType(types.int64), args[0])
 
 
+@cuda_decl_registry.register_global(abs)
+class MaskedScalarAbsoluteValue(AbstractTemplate):
+    """
+    Typing for the builtin function abs. Returns the same
+    type as input except for boolean values which are converted
+    to integer.
+
+    This follows the expected result from the builtin abs function
+    which differs from numpy - np.abs returns a bool whereas abs
+    itself performs the cast.
```

Comment:
I am not sure we want this special-casing. `abs(pd.Series([0, 1], dtype=bool))` produces a series with dtype bool.

Reply:
True, but we're trying to mimic the behavior of pandas, which executes the real `abs` function in a loop over the scalars in the data:

```python
import pandas as pd
sr = pd.Series([0, 1], dtype='bool')
sr.apply(lambda x: abs(x))
```
this gives me 
```python
0    0
1    1
dtype: int64
```

Reply:
Oh, sure. I mean this is horrible, but so it goes.